src/types/basic/abstract.ts

```diff
@@ -165,7 +166,7 @@ export class BasicType<T> {
   /**
    * Convert to JSON-serializable object
    */
-  toJson(value: T): Json {
+  toJson(value: T, options?: IJsonOptions): Json {
```

Comment:
should fromJson also have these options?

Reply:
Probably, but none of the basic types are using it. This was just to fix some type incompatibility

Reply:
Can you add it for completeness if you're adding it to `toJson` here/

I'd love to refactor to merge BasicType and CompositeType at some point.

Reply:
Yeah we could have a lot more typesafety if we manage to merge them